loader/unknown_ext_chain.c

```diff
@@ -33,33 +33,40 @@
 #pragma GCC optimize(3)  // force gcc to use tail-calls
 #endif
 
+// clang 13 has a tail call attribute for return statements
+#if defined(__clang__) && __clang_major__ >= 13
+#define TAIL_CALL __attribute__((musttail))
+#else
+#define TAIL_CALL
+#endif
+
 // Trampoline function macro for unknown physical device extension command.
-#define PhysDevExtTramp(num)                                                              \
-    VKAPI_ATTR void VKAPI_CALL vkPhysDevExtTramp##num(VkPhysicalDevice physical_device) { \
-        const struct loader_instance_dispatch_table *disp;                                \
-        disp = loader_get_instance_dispatch(physical_device);                             \
-        disp->phys_dev_ext[num](loader_unwrap_physical_device(physical_device));          \
+#define PhysDevExtTramp(num)                                                                      \
+    VKAPI_ATTR VkResult VKAPI_CALL vkPhysDevExtTramp##num(VkPhysicalDevice physical_device) {     \
```

Comment:
I thought there was a reason we returned `void` here.  Either most calls don't return something or somethiing else.  Perhaps @charles-lunarg can chime in when he's back online.

Reply:
Good job finding the sketchiest part of this change! I had to update the loader signature to match the function pointer signature because clang enforces that they match when using `__attribute__((musttail))`

https://github.com/KhronosGroup/Vulkan-Headers/blob/18963a6cc03fe15e3785d353dea6a1ff95115a5e/include/vulkan/vk_layer.h#L73-L74

I was able to convince myself that this is correct but would appreciate some extra eyes.